packages/contracts-bedrock/test/safe/SaferSafes.t.sol

```diff
@@ -144,3 +158,260 @@ contract SaferSafes_Uncategorized_Test is SaferSafes_TestInit {
+        SafeTestLib.enableModule(safeInstance, address(livenessModule2));
+        _enableModule(safeInstance, CHALLENGE_PERIOD, fallbackOwner);
+
+        // Enable 1 more module to bring us to 100 modules.
+        SafeTestLib.enableModule(safeInstance, address(makeAddr("module100")));
+
+        // Start a challenge
+        vm.prank(fallbackOwner);
+        livenessModule2.challenge(address(safeInstance.safe));
+
+        // Warp past challenge period
+        vm.warp(block.timestamp + CHALLENGE_PERIOD + 1);
+
+        // Execute ownership transfer
+        vm.expectEmit(true, true, true, true);
+        emit ChallengeSucceeded(address(safeInstance.safe), fallbackOwner);
+
+        vm.prank(fallbackOwner);
+        livenessModule2.changeOwnershipToFallback(address(safeInstance.safe));
+
+        // Ensure that the call is below a safe gas limit. The EIP-7825 limit is 16,777,216, so 12M is a safe limit.
+        assertLt(vm.lastCallGas().gasTotalUsed, 12_000_000);
+
+        // Verify module is disabled
+        assertFalse(ModuleManager(safeInstance.safe).isModuleEnabled(address(livenessModule2)));
+        // Verify extra modules are still enabled
+        (address[] memory modules,) = ModuleManager(safeInstance.safe).getModulesPaginated(address(1), 1000);
+        assertEq(modules.length, 99);
+
+        _assertOwnershipChanged(address(safeInstance.safe));
+    }
+
+    function test_changeOwnershipToFallback_withOtherGuard_succeeds() external {
+        _enableModule(safeInstance, CHALLENGE_PERIOD, fallbackOwner);
+        // Create a mock guard
+        address dummyGuard = makeAddr("dummyGuard");
+        vm.mockCall(
+            dummyGuard,
+            abi.encodeCall(
+                IGuard.checkTransaction,
+                (address(0), 0, "", Enum.Operation.Call, 0, 0, 0, address(0), payable(address(0)), "", address(0))
+            ),
+            ""
+        );
+        vm.mockCall(dummyGuard, abi.encodeCall(IGuard.checkAfterExecution, (bytes32(0), false)), "");
+
+        // Enable the mock guard on the Safe
+        SafeTestLib.execTransaction(
+            safeInstance,
+            address(safeInstance.safe),
+            0,
+            abi.encodeCall(GuardManager.setGuard, (dummyGuard)),
+            Enum.Operation.Call
+        );
+
+        // Start a challenge
+        vm.prank(fallbackOwner);
+        livenessModule2.challenge(address(safeInstance.safe));
+
+        // Warp past challenge period
+        vm.warp(block.timestamp + CHALLENGE_PERIOD + 1);
+
+        // Execute ownership transfer
+        vm.expectEmit(true, true, true, true);
+        emit ChallengeSucceeded(address(safeInstance.safe), fallbackOwner);
+
+        vm.prank(fallbackOwner);
+        livenessModule2.changeOwnershipToFallback(address(safeInstance.safe));
+
+        // These checks include ensuring that the guard is deactivated
+        _assertOwnershipChanged(address(safeInstance.safe));
+    }
+
```

Comment:
below here the tests are all pre-existing, just moved from the LivenessModule2 suite.